src/pages/VideoPlayer/VideoPlayer.jsx

```diff
@@ -21,20 +28,92 @@ const Container = styled(Grid)({
 });
 function VideoPlayer() {
     const location = useLocation();
+    const history = useHistory();
+
+    const { video, from } = location.state ? location.state : {};
+
+    const { authenticated } = useAuth();
+    const currentFavList = storage.get(FAVORITES_LIST);
+    const currentLatList = storage.get(WATCH_LATER_LIST);
+    const [isFavorite, setIsFavorite] = useState(video ? currentFavList.find(f => f.id.videoId === video.id.videoId) : [])
```

Comment:
the name of this variable makes you think of a boolean but the value is an array, a better name could improve readability